src/openzaak/utils/filters.py

```diff
@@ -49,3 +51,36 @@ def __init__(self, *args, **kwargs):
 
     def filter(self, qs, value):
         return qs
+
+
+class KeyValueFilter(filters.CharFilter):
+
+    def __init__(self, key_field_name, value_field_name, *args, **kwargs):
+        kwargs.setdefault(
+            "validators", [RegexValidator(r"^([^:\[\]]+):([^:\[\]]+)$")]
+        )  # key:value where the key and value cannot contain `[`, `]` or `:`
```

Comment:
Is it possible to specify this as a class variable to make sure it's always applied? Because if we reuse this filter and pass extra validators, this one won't be applied I think